### ISSUE TITLE
What should my confidence level be in the output of IDfix?

### ISSUE THREAD

**Issue by MDAMatt** [microsoft/idfix#18]
### Category
- [ ] Enhancement
- [ ] Bug
- [x] Question
- [ ] Documentation gap/issue

### Version
Please specify what version of IdFix you are using: [ 2.2.0.0 ]

###  Question
What level of confidence can admins expect from the recommendations delivered via this tool. Are the possible recommendations the tool presents back reviewed by the AD team?

### Observed Behavior
Running the IDFix 2.2.0.0 in my environment returns several recommendations for removing Parentheses from the x500 addreses, which is being addressed in #16 . It also makes suggestions about populating blank displayName attributes for ADFS Certificate Sharing Containers. One would assume that ADFS configured the sharing containers in the way that is supported and expected by AD and ADFS. I interpret this as a bug in the IDfix or that we encountered a bug in the ADFS setup somewhere in it's history, but can't find any documentation supporting the need for this change.

### Steps to Reproduce
Run IDfix with ADFS in environment

**Comment by patrick-rodgers** [comment]
We believe the checks and recommendations to be correct, but things change in how AAD works or things may be wrong in the tool. It sounds like you are being caught by the rule saying displayname shouldn't be empty where perhaps that is allowed for certain object types?

**Comment by MDAMatt** [comment]
> We believe the checks and recommendations to be correct, but things change in how AAD works or things may be wrong in the tool. It sounds like you are being caught by the rule saying displayname shouldn't be empty where perhaps that is allowed for certain object types?

thanks @patrick-rodgers. Considering this is an official Microsoft tool that is referenced in the official documentation as a pre-requisite for configuring the Hybrid AAD setup, I would expect tighter QC on making recommendations for changes to the directory. If AD or AAD functionality changes, one would expect the tool that's used to diagnose issues between the two of them would be updated as well. I could understand if this were a third party open source tool, but it's a Microsoft developed tool that is frequently referenced in official Microsoft documentation. It should be held to the same standards.

**Comment by danvailcolorado** [comment]
thanks @patrick-rodgers.  I am having the same exact same issues with 2.2.0.0.  I am currently using Azure AD Connect to upload a subset of users into Azure and was cleaning up problems reported by a p[revious version of IDFix.  I was down to just a handful of issues and then noticed I was using an older version of IDFix.  I upgraded to 2.2.0.0 and then reran it and it added over 150 addtional for removing Parentheses from the x500 addreses.  I was not paying attention to the subset of users and made a couple of the edits in IDFix, but then noticed that something was fighting me and changing the attribute in AD back.  I did it again and confirmed the attributes in AD, confirm the change was done as intended in AD, and then confirmed that Azure aD Connect was changing it back to the value that IDFix is reporting as a problem.  The Service Sync Manager on Azure AD Connect is not reporting any issues.  My version of Azure AD Connect is 1.4 and a few months old, so I am going to update it to 1.5, but if IDFix continues to have same issue after that I will assume it is a bug.
